model/restaurantModel.go

```diff
@@ -0,0 +1,21 @@
+package model
+
+import (
+	"gorm.io/gorm"
+)
+
+type Restaurant struct {
+	gorm.Model
+	Name          string   `json:"name"`
+	Address       string   `json:"address"`
+	HasPickup     bool     `json:"has_pickup"`
+	HasDelivery   bool     `json:"has_delivery"`
+	IsVegeterian  bool     `json:"is_vegeterian"`
+	IsVegan       bool     `json:"is_vegan"`
+	IsPescaterian bool     `json:"is_pescaterian"`
+	CuisineTags   []string `json:"cuisines_tags"`
+	AverageRating float32  `json:"average_rating"`
+	Reviews       string   `json:"reviews"`
+	Hours         string   `json:"hours"`
```

Comment:
(for all), we dont want the `json:--` as we will not be serializing directly from json into the db model. We would want separation of models, from the view (db) layer to the external (api) layer.

Flow would be like this:
User sends serialized request (in JSON) -> Gin API deserializes into go struct (using :json) -> it is passed to a mapper method (convert extern struct to view struct) -> Gorm puts it in the db


Reason is we dont want to couple our API model with our view model. One being we dont want to expose our full DB model to the client by default (some fields we want to keep internal). Another is that our API models will soon not match our DB model, for a singe GET call response this would work, but as we add API's that return n results and modification API's, we would need to define models specific to the API itself.